src/common/NetworkManager.ts

```diff
@@ -74,6 +74,10 @@ export class NetworkManager extends EventEmitter {
     string,
     Protocol.Network.RequestWillBeSentEvent
   >();
+  _requestIdToRequestPausedEvent = new Map<
```

Comment:
Could we add a comment explaining how this works? I.e., which events under which circumstances are expected to be stored here.

Reply:
See https://github.com/puppeteer/puppeteer/pull/7060#discussion_r608784254

Events will be added to `_requestIdToRequestWillBeSentEvent` if `_onRequestWillBeSent` runs before `_onRequestPaused`.
Events will be added to `_requestIdToRequestPausedEvent` if `_onRequestPaused` runs before `_onRequestWillBeSent`.

Reply:
I'll add a comment for both explaining that the event order is variable and that we need both for `_onRequest`, so we store whichever event arrives first.